File: app/src/main/java/ru/samtakoy/core/domain/CoursesPlanner.java
package ru.samtakoy.core.domain;

import androidx.annotation.Nullable;

import java.util.List;

import ru.samtakoy.core.data.local.database.room.entities.elements.Schedule;

public interface CoursesPlanner {


    // проверить позднее - есть ли начатые незавершенные обучения/повторения и показать нотификацию
    void planUncompletedTasksChecking();

    void reScheduleLearnCourses();

    void planAdditionalCards(Long qPackId, List<Long> errorCardIds, Schedule schedule);

    void planAdditionalCards(Long qPackId, @Nullable String subTitle, List<Long> cardIds, Schedule restSchedule);
}

File: app/src/main/java/ru/samtakoy/features/learncourse/domain/CoursesPlanner.kt
package ru.samtakoy.features.learncourse.domain

import ru.samtakoy.features.learncourse.domain.model.schedule.Schedule

interface CoursesPlanner {
    // проверить позднее - есть ли начатые незавершенные обучения/повторения и показать нотификацию
    fun planUncompletedTasksChecking()

    fun reScheduleLearnCourses()

    fun planAdditionalCards(qPackId: Long, errorCardIds: List<Long>, schedule: Schedule)

    fun planAdditionalCards(qPackId: Long, subTitle: String?, cardIds: List<Long>, restSchedule: Schedule)
}
